### PR TITLE
ISPN-14490 Add blocking scheduled tasks to BlockingManager

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-14490

### CODE REVIEW

File: core/src/main/java/org/infinispan/util/concurrent/BlockingManager.java [hunk 1]
```diff
@@ -247,12 +248,46 @@ <V> CompletionStage<V> whenCompleteBlocking(CompletionStage<V> stage,
...
+    * <p>
+    * Unlike other methods in this interface, the submitting thread does not impact this method's behavior.
+    *
+    * @param runnable blocking operation that runs some code.
+    * @param delay the time from now to delay execution
+    * @param unit the time unit of the delay parameter
+    * @param traceId an identifier that can be used to tell in a trace when an operation moves between threads.
+    * @return a stage that is completed after the runnable is done or throws an exception.
+    */
+   default ScheduledCompletableStage<Void> scheduleRunBlocking(Runnable runnable, long delay, TimeUnit unit, Object traceId) {
```
Comment (fax4ever) [inline core/src/main/java/org/infinispan/util/concurrent/BlockingManager.java:270]: Is there a way to return the `ScheduledFuture` instead?

Comment (fax4ever) [reply core/src/main/java/org/infinispan/util/concurrent/BlockingManager.java:270]: I tried to do that here => https://github.com/infinispan/infinispan/pull/10606/commits/5c510feea2dd2022b0e829ad27d3eb057146217c
Is it safe to expose it?

Comment (wburns) [reply core/src/main/java/org/infinispan/util/concurrent/BlockingManager.java:270]: I mentioned it here https://github.com/infinispan/infinispan/pull/10606#discussion_r1089058467. But I would say lets keep this commit as is and you can change it in your PR so we know what it was changed for specifically.

**Review by jabolina — APPROVED** [review]

File: core/src/main/java/org/infinispan/util/concurrent/BlockingManagerImpl.java [hunk 2]
```diff
@@ -353,6 +362,64 @@ public <V> CompletionStage<V> supply(Supplier<V> supplier, Object traceId) {
...
+         CompletableFuture.supplyAsync(() -> {
+                  log.tracef("Running blocking supply operation %s", traceId);
+                  return supplier.get();
+               }, blockingExecutor)
+               .whenComplete((v, t) -> {
+                  if (t != null) {
+                     log.tracef("Operation %s completed exceptionally with message %s", traceId, t.getMessage());
+                     nonBlockingManager.completeExceptionally(this, t);
+                  } else {
+                     log.tracef("Operation %s completed normally", traceId, t.getMessage());
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/util/concurrent/BlockingManagerImpl.java:415]: `t` is null:
```suggestion
                     log.tracef("Operation %s completed normally", traceId);
```

**Review by pruivo — CHANGES_REQUESTED** [review]
LGTM, just a small typo :+1:

File: core/src/main/java/org/infinispan/util/concurrent/BlockingManagerImpl.java (hunk 2 above)
Comment (wburns) [reply core/src/main/java/org/infinispan/util/concurrent/BlockingManagerImpl.java:415]: Copy paste taking me down :)

**Comment by wburns** [comment]
Updated

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
merged! thanks @wburns !

**Pull request merged**